### ISSUE TITLE
[5.1] Unable to use foreign keys when using SQLite

### ISSUE THREAD

**Issue by kidshenlong** [laravel/framework#10190]
A user has a collection of related items. For example books and authors. When a user deletes an author, the author's books should also be deleted.

SQLite databases have this behaviour disabled by default but in previous versions of Laravel (4.*) it was possible to precede requests with a raw SQL statement ```'PRAGMA foreign_keys=1'``` to enable this behaviour. This statement does not work in the latest version of Laravel.

To verify this issue I created a suite of simple tests. These tests passed when running against a MySQL DB they also worked when ported back to Laravel 4.

**Tl;dr Deleting parent objects does not cascade to children when using SQLite as a DB even when enabling Foreign Keys.**

**Comment by jobrios** [comment]
I'm not experiencing this problem with Laravel 5.1.16 (SQLite 3.7.7). I executed the `PRAGMA foreign_keys = ON` statement in `AppServiceProvider#boot` as shown [here](http://ambracode.com/index/show/24257). When I deleted the parent in `artisan tinker`, the children were deleted.

**Comment by narsariamanu** [comment]
I tried the same solution as suggested above in Laravel 5.2.39 and SQLite 3.13.0, it didn't work. I executed the PRAGMA foreign_keys = ON statement in AppServiceProvider#boot as shown [here](http://ambracode.com/index/show/24257). It didn't work, the children were not deleted. Any help would be appreciated.

**Comment by dakira** [comment]
@narsariamanu I just came across this problem. Issuing

    \DB::statement(\DB::raw('PRAGMA foreign_keys = ON;'));

directly before the delete or in AppServiceProvider#boot works perfectly for me.

**Comment by jago86** [comment]
I'm in Laravel 5.5 and with this solution I have a strange behaivor. When I try to ->attach() ONE model to another (many to many relationship): *PDOException: SQLSTATE[HY000]: General error: 1 no such table: main.cv*

 **cvs** is my table and I'm trying 
```$this->cvs()->attach($cv);```

It seems like are trying to get a table named **'main.cv'** instead of **'cvs'**
Someone has any similar problem?

**Comment by dakira** [comment]
@rajesh-parmar works fine with 5.7. As per the official documentation just use this:

    \Schema::enableForeignKeyConstraints();
